src-clean/server.js

```diff
@@ -4,63 +4,70 @@ import serialize from 'serialize-javascript'
 import styleSheet from 'styled-components/lib/models/StyleSheet'
 import { renderToString, renderToStaticMarkup } from 'react-dom/server'
 import { Provider } from 'react-redux'
-import { createMemoryHistory, RouterContext, match } from 'react-router'
-import { syncHistoryWithStore } from 'react-router-redux'
+import { matchRoutes } from 'react-router-config'
+import { ConnectedRouter, push } from 'react-router-redux'
+import createHistory from 'history/createMemoryHistory'
 import { Router } from 'express'
 import express from 'services/express'
-import routes from 'routes'
+import AppRoutes, { routes } from 'routes'
 import configureStore from 'store/configure'
 import { env, port, ip, basename } from 'config'
 import Html from 'components/Html'
 
 const router = new Router()
 
-router.use((req, res, next) => {
+router.use((req, res) => {
   if (env === 'development') {
     global.webpackIsomorphicTools.refresh()
   }
 
   const location = req.url.replace(basename, '')
-  const memoryHistory = createMemoryHistory({ basename })
-  const store = configureStore({}, memoryHistory)
-  const history = syncHistoryWithStore(memoryHistory, store)
+  const history = createHistory()
+  const store = configureStore({}, history)
 
-  match({ history, routes, location }, (error, redirectLocation, renderProps) => {
-    if (redirectLocation) {
-      res.redirect(redirectLocation.pathname + redirectLocation.search)
-    }
+  store.dispatch(push(req.url))
 
-    if (error || !renderProps) {
-      return next(error)
-    }
+  const context = {}
 
-    const fetchData = () => new Promise((resolve, reject) => {
-      const method = req.method.toLowerCase()
-      const { params, location, components } = renderProps
-      const promises = []
-
-      components.forEach((component) => {
-        if (component) {
-          while (component && !component[method]) {
-            // eslint-disable-next-line no-param-reassign
-            component = component.WrappedComponent
-          }
-          component &&
-          component[method] &&
-          promises.push(component[method]({ req, res, params, location, store }))
+  const fetchData = () => new Promise((resolve, reject) => {
```

Comment:
@kybarg Sorry for the delay!  This was not an issue with the upgrade.  **The problem existed before the upgrade.**  It has to do with the front end adding identical keys when creating a new post.  I will address this after **React-Router-4** is merged :)